http3/client.go

```diff
@@ -127,15 +129,45 @@ func (c *client) setupSession() error {
 		return err
 	}
 	buf := &bytes.Buffer{}
-	// write the type byte
-	buf.Write([]byte{0x0})
+	utils.WriteVarInt(buf, streamTypeControlStream)
 	// send the SETTINGS frame
 	(&settingsFrame{}).Write(buf)
-	if _, err := str.Write(buf.Bytes()); err != nil {
-		return err
-	}
+	_, err = str.Write(buf.Bytes())
```

Comment:
Ignoring short writes here?

Reply:
Not ignoring it. But there's no temporary `Write` errors.